libs/azure-storage/README.md

```diff
@@ -1,9 +1,147 @@
+- `AzureBlobStorageLoader` - Loads LangChain document objects from Azure Blob Storage
 
 ## Installation
 
 ```bash
 pip install -U langchain-azure-storage
 ```
+
+## Document Loader Examples
+
+### Load from container
+Below shows how to load documents from all blobs in a given container in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+The example below shows how to load documents from blobs in a container with a given prefix:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    prefix="test",
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load from a list of blobs
+Below shows how to load documents from a list of blobs in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    blob_names=["blob-1", "blob-2", "blob-3"],
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load asynchronously
+Below shows how to load documents asynchronously which can be done by using either `aload()` or `alazy_load()`:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+async def main():
+    loader = AzureBlobStorageLoader(
+        account_url=<your_account_url>,
+        container_name=<your_container_name>,
+    )
+
+    async for doc in loader.alazy_load():
+        print(doc.page_content)
+```
+
+
+### Override loader
+Below shows how to override the default loader used to parse blobs:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_unstructured import UnstructuredLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    loader_factory=UnstructuredLoader,
+)
+```
+
+To provide additional configuration, you can define a callable that returns an instantiated document loader as shown below:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_unstructured import UnstructuredLoader
+
+
+def loader_factory(file_path: str) -> UnstructuredLoader:
+    return UnstructuredLoader(
+        file_path,
+        mode="by_title",  # Custom configuration
+        strategy="fast",  # Custom configuration
+    )
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    loader_factory=loader_factory,
+)
+```
+
+### Override credentials
```

Comment:
For this section, let's:
1. Move it just below the list by blob name section to keep it close to section related to azure configuration of the loader
2. Rename it to `Override default credentials` to make it more explicit what this example would be overriding.